DuggaSys/microservices/monolithic-to-microservices.md

```diff
@@ -0,0 +1,123 @@
+
+Disclaimer, this guide serves as a guideline and can not be followed 100% since the structure of all files differ. It's recommended to test after each step which include code changing to so its faster to spot if something goes wrong. When this guide was written no service using retrive information was used, write guide at step 5.
+***
+##Setting up file & folder:
+***
+
+###Step 1
+Locate the folder microservices:
+***DuggaSys -> microservices***
+
+###Step 2
+If a folder named after the monolithic service file exists skip to step 3.
+Else if it does not exist, create a folder named after the service file, for example codeviewerService.php(monolithic) becomes codeviewerService(Folder).
+
+###Step 3
+Create a file with name following the naming convention in Potential_micro_services:
+***Backend-models -> microservices -> duggaSys services***
+
+**Example:**
+| codeviewerService |
+| --- |
+| editBoxTitle_ms.php |
+| createBox_ms.php |
+
+***
+##Converting monolithic to microservice:
+###Step 1
+Copy the entire service file content from the monolithic php to your new made, the purpose of this is to simmer down 
+
+###Step 2
+Remove all the “opt”s that are not relevant to you. If your function is an else if, make it into an If statement. 
+Once all irrelevant code is removed test if it still works (see “How to test: section)
+
+###Step 3
+Remove global variables that your functions do not need.
+For visuall representation, check **codeviewerService** and the microservice **editBoxTitle_ms**
+
+###Step 4
+Add the following include in code example and call getUid(); once the code is added remove code that belongs in getUid(see code example:getUid)
+#####Codeexample: Include
+```
+include ('../shared_microservices/getUid_ms.php');
+getUid();
+```
+#####Codeexample: getUid
+```
```

Comment:
Suggestion: writing `php` directly after the third backtick gives the code syntax highlighting.
![image](https://github.com/HGustavs/LenaSYS/assets/129070844/4a64bcf1-3f34-4e00-9f19-a463471c34a4)


```php
// code
include ('../shared_microservices/getUid_ms.php');
getUid();
````

Reply:
This goes for all code examples